packages/react-native-web/src/exports/Image/index.js

```diff
@@ -191,33 +236,24 @@ const Image: React.AbstractComponent<
     }
   }
 
-  const [state, updateState] = React.useState(() => {
-    const uri = resolveAssetUri(source);
-    if (uri != null) {
-      const isLoaded = ImageLoader.has(uri);
-      if (isLoaded) {
-        return LOADED;
-      }
-    }
-    return IDLE;
-  });
+  const imageLoadingProps = { onLoad, onLoadStart, onLoadEnd, onError };
+
+  const fallbackSource = useSource(imageLoadingProps, defaultSource);
+  const mainSource = useSource(imageLoadingProps, source);
```

Comment:
Extracted the `useEffect` hook here in order to load both `defaultSource` and `source` using the same logic (and supporting source headers)

Any dependencies were moved to the hook